src/server/auth/server/testing/auth_test.go

```diff
@@ -983,14 +983,14 @@ func TestStopAndDeletePipeline(t *testing.T) {
 		buildBindings(alice, auth.RepoOwnerRole, bob, auth.RepoWriterRole, pl(pipeline), auth.RepoWriterRole),
 		getRepoRoleBinding(t, aliceClient, pipeline))
 
-	// bob still can't stop or delete alice's pipeline
-	err = bobClient.StopPipeline(pipeline)
+	// bob can now stop the pipeline, but can't start or delete it
```

Comment:
Is this comment accurate? I thought Bob could delete this pipeline, and we're adding the ability to stop the pipeline so it can be stopped before deleting? 

For symmetry, should bob also be able to restart the stopped pipeline? It kind of feels like the only time we should care about the input repos is at pipeline creation/update time.

Reply:
My goal was to not change the permissions for deleting a pipeline at all. I realize now that I was still thinking about the old auth model, where having `delete` permission implied `write` permission, but in principle someone could create role bindings now which make that not the case.

The main intended auth change was to not require input repo read access to stop a pipeline. I considered your symmetry point - we currently have the same issue where being able to delete a pipeline does not mean you can recreate it. I agree with you that input repo read access seems like a more strict requirement than makes sense in many of these cases.

The two solutions I'm thinking of are just ignoring auth errors on the stop attempt during delete, which puts aside the issue of the proper permissions for `StopPipeline`, or add a new explicit stop/start pipeline operation type with the desired semantics. Should probably do both, as it would be bad for a delete to fail for lack of write access, as bizarre as the perm combo seems to me.

Reply:
Yeah I'm thinking let's get away from reusing the operation types and be explicit about each one - it might be a little more verbose but then this kind of issue will be immediately apparent :)

Reply:
I think this comment is still inaccurate but otherwise the behaviour looks good 👍

Reply:
Ah whoops, thanks! fixed things up